ci/test_wheel_distributed_ucxx.sh

```diff
@@ -19,6 +19,12 @@ python -m pip install ./local-ucxx-dep/ucxx*.whl
 # echo to expand wildcard before adding `[extra]` requires for pip
 python -m pip install $(echo ./dist/${PROJECT_NAME}*.whl)[test]
 
+# TODO: We need distributed installed in developer mode to provide test utils,
+# we still need to match to the `rapids-dask-dependency` version.
+rapids-logger "Install Distributed in developer mode"
+git clone https://github.com/dask/distributed /tmp/distributed
+python -m pip install -e /tmp/distributed
```

Comment:
@vyasr we don't need to address this immediately, but do you think it would work if we had optional dependencies for tests in https://github.com/rapidsai/rapids-dask-dependency/blob/ac821e6a3e396340f65fe79dc834f5b711d3b0cb/pip/rapids-dask-dependency/pyproject.toml#L14-L17 , where we essentially `pip install -e` the packages? I don't know if that's possible -- and I think it isn't -- but would like to know if you have any ideas how this could be done. We would also need something similar for conda packages, I currently [`pip install -e` in conda CI as well](https://github.com/rapidsai/ucxx/blob/94a19fde0e718ab3af18e1afc5ba79988dee9fdd/ci/test_python.sh#L99-L100) which isn't a good long-term solution.

Reply:
I had [a question](https://github.com/rapidsai/ucxx/pull/167#discussion_r1470174337) about this as well. Let's discuss in this thread.

Reply:
Also adding here what I wrote in the other thread:
> Yes, we need some testing dependencies from distributed, specifically https://github.com/rapidsai/ucxx/blob/94a19fde0e718ab3af18e1afc5ba79988dee9fdd/python/distributed-ucxx/distributed_ucxx/tests/test_ucxx.py#L173
> I don't know of a good solution for this problem either though and I very much don't want to copy-paste the code in here because that will eventually cause us to get out-of-sync with Distributed code and essentially mean we're not testing it properly anymore.

Reply:
You're right that they intentionally do not ship because they are only used in that particular test file. I very much wanted to avoid having to modify things there, the file is reasonably large with many functions that would need to move and we would have to bear the burden of properly documenting/annotating those functions that I'm not even familiar with internals, and eventually we would also need to bear the burden of some maintenance, which we currently don't have the bandwidth to. If we can't find a reasonably simple solution to install Distributed in dev mode, I think I'd prefer to limit or skip that test for now.

Reply:
I'd rather limit or skip. This logic:
> I very much don't want to copy-paste the code in here because that will eventually cause us to get out-of-sync with Distributed code and essentially mean we're not testing it properly anymore.

feels directly contradictory to 
> we would have to bear the burden of properly documenting/annotating those functions that I'm not even familiar with internals, and eventually we would also need to bear the burden of some maintenance, which we currently don't have the bandwidth to

If we don't want to bear the maintenance burden then I'd much rather not try and take on some partial maintenance burden via a hacky installation of distributed here that forces us to stay partially in sync.

Reply:
I don't see where the contradiction as the statements are talking about two different aspects:
1. Copy-pasting those function _in distributed-ucxx_ means they will eventually become out-of-sync with the actual implementation from Distributed, which is the source-of-truth;
2. The maintenance burden I'm referring to is the maintenance burden of those functions _in Distributed_ as we will now have "some authorship" because we're requesting they live as part of distributed packages and will have to document and properly type-annotate them, etc.

IOW, we still want to use them here _for testing purposes only_ but I'd very much avoid touching much of Distributed code if we don't absolutely need to.

Reply:
Right, but right now we implicitly incur part of the maintenance burden of 2 because if dask decides to change the behavior of that function in some way we'll see it in our test. We're not maintaining the function, but we are effectively signing up to keep track of an implementation detail of dask's tests. Conversely, if it was public there would be a higher maintenance burden (because we'd have to maintain docs/annotations/etc as you say) but we would also be less likely to be broken without warning (of course, given dask's low stability it could still happen, but it reduces the odds). The current approach also makes it harder for devs to understand why the tests fail if they don't know that they need to download the source of distributed to access this functionality.

In any case, I don't think there is a good solution for installing the editable version from source. Every solution I can think of is equally hacky, so for now I think we can leave this as is.

Reply:
I think the split you proposed for getting extra coverage in CI. Let's make that change in a follow-up PR where you apply it to both conda and pip testing.